compiler/src/main/scala/org/scalajs/core/compiler/util/VarBox.scala

```diff
@@ -0,0 +1,9 @@
+package org.scalajs.core.compiler.util
+
+import language.implicitConversions
+
+class VarBox[T](var value: T)
+
+object VarBox {
+  implicit def toValue[T](box: VarBox[T]): T = box.value
```

Comment:
This implicit bothers me, because `value` is fundamentally mutable. I think it should be removed, as well as `toValueFromBox`. And instead write `.value` explicitly when we want to read the value.